contracts/root/withdrawManager/WithdrawManager.sol

```diff
@@ -294,22 +291,50 @@ contract WithdrawManager is WithdrawManagerStorage, IWithdrawManager {
         }
       }
 
-      emit Withdraw(exitor, _token, amountOrNft);
-
-      // Delete owner but keep amount to prevent another exit from the same UTXO.
-      // delete exits[exitId].owner;
-      exitQueue.delMin();
+      emit Withdraw(exitId, exitor, _token, amountOrNft);
 
-      if (currentExit.isRegularExit) {
-        // delete current exit for the owner, so they can do another one in the future
-        delete ownerExits[keccak256(abi.encodePacked(_token, currentExit.owner))];
-      } else {
+      if (!currentExit.isRegularExit) {
         // return the bond amount if this was a MoreVp style exit
         address(uint160(exitor)).transfer(BOND_AMOUNT);
       }
     }
   }
 
+  function getExitId(address exitor, uint256 priority)
+    internal
+    pure
+    returns (uint256 exitId)
+  {
+    // priority queue expects 128 most significant bits to be zero
+    exitId = uint256(keccak256(abi.encodePacked(exitor, priority))) >> 128;
+  }
+
+  function getKey(address token, address exitor, uint256 amountOrToken)
+    internal
+    view
+    returns (bytes32 key)
+  {
+    if (registry.isERC721(token)) {
+      key = keccak256(abi.encodePacked(token, exitor, amountOrToken));
+    } else {
+      // validate amount
+      require(amountOrToken > 0, "CANNOT_EXIT_ZERO_AMOUNTS");
+      key = keccak256(abi.encodePacked(token, exitor));
+    }
+  }
+
+  function getExitableWindow()
+    internal
+    returns(uint256)
+  {
+    // Reset exitWindow if it is nearing its end
```

Comment:
Please explain.

Reply:
Discussed offline, not the best solution.